Add disconnected screen for when livekit disconnects from the call

dbkr opened the pull request:
eg. when it gives up retrying to connect

Also refactors the rageshake button into a dedicated component that we can just put wherever we want.

Not the prettiest screen:
![Jul-20-2023 17-54-26](https://github.com/vector-im/element-call/assets/986903/f9341a84-3d34-459b-82e2-1ecf9cf43389)

Fixes https://github.com/vector-im/element-call/issues/1268

dbkr committed:
Add disconnected screen for when livekit disconnects from the call

eg. when it gives up retrying to connect

Also refactors the rageshake button into a dedicated component that
we can just put wherever we want.

Fixes https://github.com/vector-im/element-call/issues/1268

dbkr committed:
Track call disconnections

dbkr committed:
i18n

robintown commented on src/room/InCallView.tsx:
```diff
@@ -188,6 +189,29 @@ export function InCallView({
+    onLeave();
+  }, [onLeave]);
+
+  useEffect(() => {
+    livekitRoom.on(RoomEvent.Disconnected, onDisconnected);
+
+    return () => {
+      livekitRoom.off(RoomEvent.Disconnected, onDisconnected);
+    };
+  }, [onDisconnected, livekitRoom]);
```
I think `useEventEmitter` would work here

robintown commented on src/settings/RageshakeButton.module.css:
```diff
@@ -0,0 +1,22 @@
+*/
+
+.rageshakeControl {
+  height: 50px;
+  width: 176px;
```
I worry that hardcoding the dimensions like this is not a good idea for i18n

dbkr replied:
Yeah, fair. I've just made it text which means the button changes size causing some jumping, but avoids the hardcoding.

robintown reviewed (commented):
Generally looks good, I'd be curious to know if this screen has been tested on mobile though, given the other ticket about the feedback screen

dbkr committed:
Add a useEventEmitterThree for livekit's event emitters and use it

dbkr committed:
Fix logs component states

Also just display text when sending which makes the button jump around
a bit but avoids hardcoding the width

dbkr committed:
Merge remote-tracking branch 'origin/livekit' into dbkr/react_to_livekit_disconnect
